### PR TITLE
Developed setting to loop search from the beginning or end of document in Find Widget

### PR DESCRIPTION
**Pull request by GustavoASC**
This PR fixes part of #84304.

Developed setting which controls whether the search automatically restarts from the beginning (or the end) when no further matches can be found.

Default: true (to mantain the same behavior as before).

image

Here you can see that, when the value of this setting is _false_, 'Previous' and 'Next' buttons are sometimes disabled in Find Widget so one cannot go back or forward to further matches.

search-without-loop

This setting is under _Find_ group of settings and then do not affect Diff Editor. Maybe we can create another setting, only for Diff Editor.

### CODE REVIEW

**Comment by GustavoASC** [comment]
@rebornix When you have a free time can you take a look on this? 😃
If you agree with these changes I will keep testing on Insiders.

After that I'll probably make a new PR developing an equivalent setting under _Diff editor_ group of settings, which will affect looping navigation on next/previous changes, according to the image below:

image

**Comment by rebornix** [comment]
It makes sense to me, thanks for your help!

**Comment by GustavoASC** [comment]
@rebornix I'm happy you appreciate! :smile:

Could you merge this into master, please? Or you'd prefer someone else giving a second opinion before doing so?

When you merge this I'll keep testing on Insiders.
Thanks in advance!

**Comment by GustavoASC** [comment]
@rebornix any news on merging this?
I need to test a little bit more on Insiders before endgame...

**Comment by rebornix** [comment]
I'll take a look in debt week of Apri.

**Comment by GustavoASC** [comment]
@rebornix thanks! :smile:

File: src/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (rebornix) [inline src/.vscode/launch.json:1]: This file is not necessary.

**Review by rebornix — CHANGES_REQUESTED** [review]
LGTM in overall. Once `src/.vscode` is removed, we can merge this PR.

File: src/.vscode/launch.json (hunk 1 above)
Comment (GustavoASC) [reply src/.vscode/launch.json:1]: Yeah, this folder was unintentionally commited. Thanks for pointing this out. I'll fix that.

**Review by rebornix — APPROVED** [review]

**Comment by rebornix** [comment]
@GustavoASC thanks for your contribution! It would be in tomorrow's Insiders.

**Comment by GustavoASC** [comment]
Hi @rebornix ! Since it's a new feature, could a GIF of this new setting be added to 1.45 release notes? There are people looking for this VSCode behavior and I'm afraid they won't have knowledge about this.

E.g.: https://github.com/microsoft/vscode/issues/46504 and https://stackoverflow.com/questions/57158475/disable-search-wrap-in-vs-code

Thanks in advance!

**Pull request merged**